nupic/research/temporal_memory.py

```diff
@@ -593,7 +597,7 @@ def adaptSegment(self, segment, activeSynapses, connections,
     @param permanenceIncrement  (float)  Amount to increment active synapses
     @param permanenceDecrement  (float)  Amount to decrement inactive synapses
     """
-    for synapse in connections.synapsesForSegment(segment):
```

Comment:
No, it's because `destroySynapse` modifies the set of synapses while it's being iterating over, so this change is necessary to avoid an error.